File: gsonpath-compiler/src/main/java/gsonpath/model/InterfaceFieldInfo.java
package gsonpath.model;

import com.squareup.javapoet.TypeName;

import javax.lang.model.element.Element;

public class InterfaceFieldInfo {
    final Element methodElement;
    final TypeName typeName;
    public final String fieldName;

    public InterfaceFieldInfo(Element methodElement, TypeName typeName, String fieldName) {
        this.methodElement = methodElement;
        this.typeName = typeName;
        this.fieldName = fieldName;
    }
}

File: gsonpath-compiler/src/main/java/gsonpath/model/InterfaceFieldInfo.kt
package gsonpath.model

import com.squareup.javapoet.TypeName

import javax.lang.model.element.Element
import javax.lang.model.type.TypeMirror

class InterfaceFieldInfo(val elementInfo: ElementInfo,
                         internal val typeName: TypeName,
                         internal val typeMirror: TypeMirror,
                         val fieldName: String,
                         val isDirectAccess: Boolean) {

    interface ElementInfo {
        val underlyingElement: Element?

        fun <T : Annotation> getAnnotation(annotationClass: Class<T>): T?

        val annotationNames: List<String>
    }

}
